src/config.ts

```diff
@@ -47,19 +47,28 @@ function validateLicenses(
   }
 }
 
-export function readConfig(): ConfigurationOptions {
-  const externalConfig = getOptionalInput('config-file')
-  if (externalConfig !== undefined) {
-    const config = readConfigFile(externalConfig)
-    // the reasoning behind reading the inline config when an external
-    // config file is provided is that we still want to allow users to
-    // pass inline options in the presence of an external config file.
-    const inlineConfig = readInlineConfig()
-    // the external config takes precedence
-    return Object.assign({}, inlineConfig, config)
-  } else {
-    return readInlineConfig()
+export async function readConfig(): Promise<ConfigurationOptions> {
+  const remoteConfigFile = getOptionalInput('remote-config-file')
+  const repoConfigFile = getOptionalInput('config-file')
+
+  const inlineConfig = readInlineConfig()
+  let remoteConfig: ConfigurationOptions = {}
+  let repoConfig: ConfigurationOptions = {}
+
+  if (remoteConfigFile !== undefined) {
+    const fileContents = readConfigFile(await getRemoteConfig(remoteConfigFile))
+    remoteConfig = {...remoteConfig, ...fileContents}
+  }
+
+  if (repoConfigFile !== undefined) {
+    const fileContents = readConfigFile(getRepoConfig(repoConfigFile))
+    repoConfig = {...repoConfig, ...fileContents}
   }
+  // the reasoning behind reading the inline config when an external
+  // config file is provided is that we still want to allow users to
+  // pass inline options in the presence of an external config file.
+  // TO DO check order of precedence
+  return {...inlineConfig, ...remoteConfig, ...repoConfig}
```

Comment:
What is the correct order of precedence here?

Reply:
We should maintain the same precedence we have today (`Object.assign({}, inlineConfig, config)`, while **allowing the users to overwrite values**.

This means that if an external/remote config file specifies a value "A", and a user overwrites that in their local repo with value "B", we'll choose "B" as the definitive answer.